src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -401,15 +401,14 @@ def recover_vault_or_hsm(cmd, client, resource_group_name=None, location=None, v
 
 
 def recover_hsm(cmd, client, hsm_name, resource_group_name, location, no_wait=False):
-    from azure.cli.core._profile import Profile
+    from azure.cli.core._profile import Profile, _TENANT_ID
 
     ManagedHsm = cmd.get_models('ManagedHsm', resource_type=ResourceType.MGMT_KEYVAULT)
     ManagedHsmSku = cmd.get_models('ManagedHsmSku', resource_type=ResourceType.MGMT_KEYVAULT)
 
     # tenantId and sku shouldn't be required
     profile = Profile(cli_ctx=cmd.cli_ctx)
-    _, _, tenant_id = profile.get_login_credentials(
-        resource=cmd.cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
+    tenant_id = profile.get_subscription()[_TENANT_ID]
```

Comment:
A little bit hacky, but `azure.cli.command_modules.role` accesses the private attribute `_config` of the client to retrieve the subscription ID:

https://github.com/Azure/azure-cli/blob/00fe3e551fbc574abfd3e099e8e15e6ecb623a06/src/azure-cli/azure/cli/command_modules/role/custom.py#L65-L66